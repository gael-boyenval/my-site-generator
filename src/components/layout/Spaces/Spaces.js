import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { spaces } from 'Styles/tokens';
import { rem } from 'Styles/utils';

const AllowedSpacesValues = Object.keys(spaces);

const DivWithSpace = (space) => ({
  all, horizontal, vertical, top, right, bottom, left,
}) => css`
  ${all &&
    css`
      ${space}: ${rem(spaces[all])};
    `};
  ${horizontal &&
    css`
      ${space}-left: ${rem(spaces[horizontal])};
      ${space}-right: ${rem(spaces[horizontal])};
    `};
  ${vertical &&
    css`
      ${space}-top: ${rem(spaces[vertical])};
      ${space}-bottom: ${rem(spaces[vertical])};
    `};
  ${top &&
    css`
      ${space}-top: ${rem(spaces[top])};
    `};
  ${right &&
    css`
      ${space}-right: ${rem(spaces[right])};
    `};
  ${bottom &&
    css`
      ${space}-bottom: ${rem(spaces[bottom])};
    `};
  ${left &&
    css`
      ${space}-left: ${rem(spaces[left])};
    `};
`;

export const Margin = styled.div`
  ${DivWithSpace('margin')};
  ${({ inline }) =>
    inline &&
    css`
      display: inline-block;
    `};
`;

export const Padding = styled.div`
  ${DivWithSpace('padding')};
  ${({ inline }) =>
    inline &&
    css`
      display: inline-block;
    `};
`;

const globalPropTypes = {
  all: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  horizontal: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  vertical: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  top: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  right: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  bottom: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  left: PropTypes.oneOfType([PropTypes.oneOf(AllowedSpacesValues), PropTypes.bool]),
  children: PropTypes.node.isRequired,
  inline: PropTypes.bool,
};

const defaultProps = {
  all: false,
  horizontal: false,
  vertical: false,
  top: false,
  right: false,
  bottom: false,
  left: false,
  inline: false,
};

Margin.propTypes = globalPropTypes;
Padding.propTypes = globalPropTypes;
Margin.defaultProps = defaultProps;
Padding.defaultProps = defaultProps;
