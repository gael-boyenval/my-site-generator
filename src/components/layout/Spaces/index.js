export { Margin, Padding } from './Spaces';
