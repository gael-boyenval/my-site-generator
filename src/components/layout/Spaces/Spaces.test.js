/* eslint-disable no-undef */
import React from 'react';

import { Margin, Padding } from './Spaces';

describe('Component Margin', () => {
  it('should be render all without crash', () => {
    const component = shallow(<Margin all="m">hello</Margin>);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should be render left without crash', () => {
    const component = shallow(<Margin left="m">hello</Margin>);

    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('Component Padding', () => {
  it('should be render without crash', () => {
    const component = shallow(<Padding all="m">hello</Padding>);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should be render left without crash', () => {
    const component = shallow(<Padding left="m">hello</Padding>);

    expect(toJson(component)).toMatchSnapshot();
  });
});
