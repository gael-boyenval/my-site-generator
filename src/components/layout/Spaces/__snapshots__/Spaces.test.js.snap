// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Margin should be render all without crash 1`] = `
.c0 {
  margin: 2.4rem;
}

<div
  className="c0"
>
  hello
</div>
`;

exports[`Component Margin should be render left without crash 1`] = `
.c0 {
  margin-left: 2.4rem;
}

<div
  className="c0"
>
  hello
</div>
`;

exports[`Component Padding should be render left without crash 1`] = `
.c0 {
  padding-left: 2.4rem;
}

<div
  className="c0"
>
  hello
</div>
`;

exports[`Component Padding should be render without crash 1`] = `
.c0 {
  padding: 2.4rem;
}

<div
  className="c0"
>
  hello
</div>
`;
